test/integration/test_utils.js

```diff
@@ -87,3 +87,20 @@ function getSelectedEditors(page) {
   });
 }
 exports.getSelectedEditors = getSelectedEditors;
+
+async function waitForEvent(page, eventName, timeout = 30000) {
+  await Promise.race([
+    // add event listener and wait for event to fire before returning
+    page.evaluate(name => {
+      return new Promise(resolve => {
+        const listener = e => {
+          document.removeEventListener(name, listener);
+          resolve();
+        };
+        document.addEventListener(name, listener);
```

Comment:
Unless I'm missing something, could this not be simplified to only the following?
```suggestion
        document.addEventListener(name, resolve, { once: true });
```

Reply:
I forgot this argument.